src/vs/workbench/contrib/comments/browser/commentsView.ts

```diff
@@ -449,11 +464,8 @@ export class CommentsPanel extends FilterViewPane implements ICommentsView {
 		}
 		if (this.isVisible()) {
 			this.hasCommentsContextKey.set(this.commentService.commentsModel.hasCommentThreads());
-
-			this.treeContainer.classList.toggle('hidden', !this.commentService.commentsModel.hasCommentThreads());
 			this.cachedFilterStats = undefined;
-			this.renderMessage();
-			this.tree?.setChildren(null, createResourceCommentsIterator(this.commentService.commentsModel));
+			this.renderComments();
```

Comment:
note: I made `renderComments` synchronous as the `Comments View filter by text` test otherwise failed, and it didn't execute any async ops anyway.